### DISCUSSION TITLE
unknown compression format: Invalid kernel

### DISCUSSION THREAD

**Discussion by gcgcgcgc** [virt-manager/virt-manager#667] [General]
I am trying to use virt-install to create a paravirtualised guest but it appears to be having an issue with the compression of the kernel. The kernel looks to be bzipped, which I thought would have been supported for a long time. Is there something I can install to get compatibility with this compression? Or is there a way that I can re-compress the kernel in a different format?

Dom0 is a Fedora 32 installation running xen version xen-4.13.1-4.fc32.x86_64
The console is text only (no graphical console installed)

The command I am using is
_virt-install --paravirt --name ext5 --vcpus 1 -f /dev/VGvmboot/ext5_root --network bridge=br100 --location http://gsl-syd.mm.fcix.net/fedora/linux/releases/40/Everything/x86_64/os/_
The output is
_Using fedora-unknown default --memory 2048
WARNING  Unable to connect to graphical console: virt-viewer not installed. Please install the 'virt-viewer' package.
WARNING  No console to launch for the guest, defaulting to --wait -1

Starting install...
Retrieving file vmlinuz...                                                                                                                                                               |  14 MB  00:00:02
Retrieving file initrd.img...                                                                                                                                                            | 142 MB  00:00:23
ERROR    internal error: libxenlight failed to create new domain 'ext5'
Domain installation does not appear to have been successful.
If it was, you can restart your domain by running:
  virsh --connect xen:///system start ext5
otherwise, please restart your installation._

The error message in /var/log/libvirt/libxl/libxl-driver.log is below
_2024-05-17 12:37:00.653+0000: xc: panic: xc_dom_bzimageloader.c:774: xc_dom_probe_bzimage_kernel: unknown compression format: Invalid kernel
2024-05-17 12:37:00.653+0000: xc: panic: xc_dom_core.c:691: xc_dom_find_loader: no loader found: Invalid kernel
2024-05-17 12:37:00.653+0000: libxl: libxl_dom.c:747:libxl__build_dom: xc_dom_parse_image failed
2024-05-17 12:37:00.653+0000: libxl: libxl_create.c:1420:domcreate_rebuild_done: Domain 29:cannot (re-)build domain: -3_

Thank you

**Comment by phrdina** [comment]
Hi, sorry for the long time to reply. This is not related to virt-manager. Looking at the error this is most likely xen related issue so you will have to ask xen community to get help.
